vllm/distributed/device_communicators/symm_mem.py

```diff
@@ -80,6 +81,11 @@ def __init__(
             self.max_size = SYMM_MEM_ALL_REDUCE_MAX_SIZES[
                 self.device_capability][self.world_size]
 
+        # allow overlapping devices in case of data parallel
+        # if torch symm mem can not initialize multicast ptr will be 0
+        if self.world_size < dist.get_world_size():
+            os.environ["TORCH_SYMM_MEM_ALLOW_OVERLAPPING_DEVICES"] = "1"
+
```

Comment:
What does do? Is it still needed?

Reply:
It was needed in case of DP. Now, we use torch symm mem only for TP Communicator, it works fine without this env setting.